tests/e2e/vsphere.go

```diff
@@ -665,6 +666,94 @@ func (vs *vSphere) getHostUUID(ctx context.Context, hostInfo string) string {
 	return ""
 }
 
+//getAllHostsIP reads the cluster name and hosts in it and returns IP array
+func (vs *vSphere) getAllHostsIP(ctx context.Context) []string {
+	var result []string
+	computeCluster := os.Getenv("CLUSTER_NAME")
+	if computeCluster == "" {
+		if guestCluster {
+			computeCluster = "compute-cluster"
+		} else if supervisorCluster {
+			computeCluster = "wcp-app-platform-sanity-cluster"
+		}
+		framework.Logf("Default cluster is chosen for test")
+	}
+
+	finder := find.NewFinder(vs.Client.Client, false)
+	dc, err := finder.Datacenter(ctx, vs.Config.Global.Datacenters)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	finder.SetDatacenter(dc)
+
+	clusterComputeResource, err := finder.ClusterComputeResourceList(ctx, "*")
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+	cluster := clusterComputeResource[0]
+	//TKG setup with NSX has edge-cluster enabled, this check is to skip that cluster
+	if !strings.Contains(cluster.Name(), computeCluster) {
+		cluster = clusterComputeResource[1]
+	}
+
+	config, err := cluster.Configuration(ctx)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	for counter := range config.VsanHostConfig {
```

Comment:
I see you are fetching `config.VsanHostConfig` in many places, why don't you move the code to fetch this into a separate routine

Reply:
Separated the common code into a another helper function